warehouse/query-core/src/main/java/datawave/query/index/lookup/CreateUidsIterator.java

```diff
@@ -80,17 +82,54 @@
     public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
         src = source;
         if (null != options) {
+            
             final String collapseOpt = options.get(COLLAPSE_UIDS);
-            if (null != collapseOpt) {
+            if (StringUtils.isNotBlank(collapseOpt)) {
                 try {
-                    collapseUids = Boolean.valueOf(collapseOpt);
+                    this.collapseUids = Boolean.parseBoolean(collapseOpt);
                 } catch (Exception e) {
-                    collapseUids = false;
+                    String errMsg = "Option [" + COLLAPSE_UIDS_THRESHOLD + "] expected a boolean, received [" + collapseOpt + "].";
+                    log.error(errMsg);
+                    throw new IllegalArgumentException(errMsg);
                 }
             }
+            
+            final String collapseThresholdOpt = options.get(COLLAPSE_UIDS_THRESHOLD);
+            if (StringUtils.isNotBlank(collapseThresholdOpt)) {
+                try {
+                    int threshold = Integer.parseInt(collapseThresholdOpt);
+                    if (threshold == -1 || threshold == Integer.MAX_VALUE) {
```

Comment:
this.collapseUidsThreshold already defaults to MAX_VALUE, so I believe you just need to handle the else-case?